test/test_helper.rb

```diff
@@ -85,6 +87,13 @@ def reconsider_gc_deferment
       end
     end
 
+    def skip_if_plugin_asked_to
+      skips = Foreman::Plugin.tests_to_skip[self.class.name].to_a
+      if skips.any?{|name| @NAME.end_with?(name)}
+        skip "Here be dragons"
```

Comment:
It should be displayed in the rake test output, in Jenkins etc.  It should state the facts, i.e. the test was disabled by plugin X.